#!/usr/bin/env python
PACKAGE = "ohm_rrl_motiondetection2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

preset_enum = gen.enum ( [  gen.const("rotating_disks",             int_t, 0, "Preset well suited for rotating disk detection"),
                            gen.const("wibbely_wobbely_strings",    int_t, 1, "Preset well suited to detect spinning strings"),
                            gen.const("far_away",                   int_t, 2, "Preset for far away target"),
                            gen.const("vibration_tolerant",         int_t, 3, "Preset to use with ground/robot vibration")
], "Presets for different scenarios")

gen.add("preset",                        int_t,  1, "Set parameters to specific values to fit the scenario", 0, 0, 3, edit_method=preset_enum)

gen.add("sobel_kernel_size",             int_t,  0, "size of the extendet sobel kernel",                0,  0, 4)
gen.add("sobel_scale",                double_t,  0, "number of frames for averaging",                   1,  0, 4)

gen.add("history",                       int_t,  0, "number of frames to compare with current frame",   2,  2, 10)
gen.add("bluring_factor",                int_t,  0, "bluring to reduce sensor noise",                   1,  0, 5)
gen.add("noise_reduction",               int_t,  0, "Reduce noise using binary threshold ",             50, 0, 100)
gen.add("merge_contours",	             int_t,  0, "dilate contours and then erode with given size",   0,  3, 20)
gen.add("min_contour_size",              int_t,  0, "filter out smaller contours to remove noise",      8,  0, 100)

gen.add("show_contours",                 bool_t, 0, "show contours as polygon in green",                False)
gen.add("show_rectangles",               bool_t, 0, "draw rectangle around detected contours",          True)

gen.add("clahe_clip_limit",		       double_t, 0, "clip limit for clahe algorithm",                   2, 0, 10)
gen.add("clahe_grid_size",		          int_t, 0,  "grid size for clahe algorithm",                   3, 3, 9)

exit(gen.generate(PACKAGE, "ohm_rrl_motiondetection2", "MotionDetection"))
